pkg/chunkenc/vartype.go

```diff
@@ -0,0 +1,257 @@
+/*
+Copyright 2018 Iguazio Systems Ltd.
+
+Licensed under the Apache License, Version 2.0 (the "License") with
+an addition restriction as set forth herein. You may not use this
+file except in compliance with the License. You may obtain a copy of
+the License at http://www.apache.org/licenses/LICENSE-2.0.
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+implied. See the License for the specific language governing
+permissions and limitations under the License.
+
+In addition, you may not use the software for any purposes that are
+illegal under applicable law, and the grant of the foregoing license
+under the Apache 2.0 license is conditioned upon your compliance with
+such restriction.
+*/
+
+package chunkenc
+
+import (
+	"encoding/binary"
+	"fmt"
+	"math"
+	"strconv"
+
+	"github.com/nuclio/logger"
+)
+
+const (
+	varTypeNil     byte = 0
+	varTypeBlob    byte = 1
+	varTypeString  byte = 2
+	varTypeBool    byte = 3
+	varTypeFloat32 byte = 4
+	varTypeFloat64 byte = 5
+	varTypeInt8    byte = 8
+	varTypeInt16   byte = 9
+	varTypeInt32   byte = 10
+	varTypeInt64   byte = 11
+)
+
+const (
+	varValueNone byte = 0
+	varValueZero byte = 1
+	varValueOnes byte = 2
+	varValueAny  byte = 3
+)
+
+// Type encoding: 6 bits for var type, 2 bits for predefined type values (e.g. None, zero, NaN, ..)
+func decodeType(t byte) (byte, byte)    { return t >> 2, t & 3 }
+func encodeType(varType, val byte) byte { return varType<<2 + val&3 }
+
+type VarChunk struct {
+	logger logger.Logger
+
+	b       []byte
+	samples uint16
+	offset  int
+}
+
+// NewVarChunk returns a new chunk with variant encoding of the given size.
```

Comment:
In comment you say "given size" please correct comment or code
Shouldn't chunk size be configurable?